### DISCUSSION TITLE
The parameter problem of addColumns

### DISCUSSION THREAD

**Discussion by xqa** [simolus3/drift#2003] [Q&A] [answered]
The parameter of addColumns is `Iterable<Expression<dynamic>>`,but the type of Column  is `GeneratedColumn<T> `,
so if I want to use addColumns, I have to add a method like `trim()`.

Below is my code:
```dart
  Stream<List<SelectItem>> choiceList() {
    final count = choices.id.count();
    final state = choices.state.max();
    final code = choices.code.trim();
    final choiceId = choices.choiceId.trim();


    final query = selectOnly(choices)
      ..addColumns([choiceId,count, state, code])
      ..groupBy([choices.code,choices.choiceId])
      ..orderBy([OrderingTerm.desc(choices.code)]);
    return query.map((row) => SelectItem(row.read(choiceId)!,row.read(code)!, row.read(count), row.read(state)!)).watch();
  }
```
Below is the generated sql log:
```
flutter: Drift: Sent SELECT TRIM(choices.choice_id) AS "c0", COUNT(choices.id) AS "c1", MAX(choices.state) AS "c2", TRIM(choices.code) AS "c3" FROM choices GROUP BY choices.code, choices.choice_id ORDER BY choices.code DESC; with args []
```
I want to use  "c0" and "c3" without TRIM,what should I do?

**Comment by simolus3** [comment, marked as answer]
`GeneratedColumn<T>` is a subtype of `Expression<dynamic>`, so you shouldn't have to change anything - you can just pass the columns to `addColumns`.

I think the warning about `GeneratedColumn` _not_ being assignable to `Expression` is a bug in the Dart analyzer. I have received numerous reports about this problem but I've never been able to reproduce it. We also use that pattern in drift:

https://github.com/simolus3/drift/blob/a927d3717a0816a8d433e4fbc6a4e785ba551c81/drift/test/database/statements/join_test.dart#L379-L381

Since it's possibly an analyzer issue, does `flutter clean` help at all?

**Reply by xqa** [reply]
Now it worked, I just restarted Android studio,it's a bug in the Dart analyzer.
